bundles/org.openhab.binding.digitalstrom/src/main/java/org/openhab/binding/digitalstrom/internal/lib/structure/devices/impl/DeviceImpl.java

```diff
@@ -90,12 +90,10 @@
     private final int maxSlatPosition = DeviceConstants.MAX_ROLLERSHUTTER;
     private final int minSlatPosition = DeviceConstants.MIN_ROLLERSHUTTER;
 
-    private final List<DeviceSensorValue> deviceSensorValues = Collections
-            .synchronizedList(new ArrayList<DeviceSensorValue>());
-    private final List<DeviceBinaryInput> deviceBinaryInputs = Collections
-            .synchronizedList(new ArrayList<DeviceBinaryInput>());
-    private final List<SensorEnum> devicePowerSensorTypes = new ArrayList<SensorEnum>();
-    private final List<SensorEnum> deviceClimateSensorTypes = new ArrayList<SensorEnum>();
+    private final List<DeviceSensorValue> deviceSensorValues = Collections.synchronizedList(new ArrayList<>());
+    private final List<DeviceBinaryInput> deviceBinaryInputs = Collections.synchronizedList(new ArrayList<>());
```

Comment:
```suggestion
    private final List<DeviceSensorValue> deviceSensorValues = new CopyOnWriteArrayList<>();
    private final List<DeviceBinaryInput> deviceBinaryInputs = new CopyOnWriteArrayList<>();
```

Reply:
Yes changing those collection changes would also be a nice improvement! For this PR I didn't want to focus on such changes. Though my fingers were sometimes itching especially when seeing outdated types like `Vector`.

Reply:
Yep, I saw those too, but definitely out of scope of this PR.